examples/job/work-queue-2/README.md

```diff
@@ -0,0 +1,294 @@
+your username and push to the Hub with the below commands. Replace
+`<username>` with your Hub username.
+
+```
+docker tag job-wq-2 <username>/job-wq-2
+docker push <username>/job-wq-2
+```
+
+If you are using [Google Container
+Registry](https://cloud.google.com/tools/container-registry/), tag
+your app image with your project ID, and push to GCR. Replace
+`<project>` with your project ID.
+
+```
+docker tag job-wq-2 gcr.io/<project>/job-wq-2
+gcloud docker push gcr.io/<project>/job-wq-2
+```
+
+## Defining a Job
+
+Here is the job definition:
+
+
+<!-- BEGIN MUNGE: EXAMPLE job.yaml -->
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Job
+metadata:
+  name: job-wq-2
+spec:
+  selector:
+    matchLabels:
+      app: job-wq-2
+  completions: 1
+  parallelism: 2
+  template:
+    metadata:
+      name: job-wq-2
+      labels:
+        app: job-wq-2
+    spec:
+      containers:
+      - name: c
+        image: gcr.io/myproject/job-wq-2
+      restartPolicy: OnFailure
+```
+
+[Download example](job.yaml?raw=true)
+<!-- END MUNGE: EXAMPLE job.yaml -->
+
+Be sure to edit the job template to
+change `gcr.io/myproject` to your own path.
+
+In this example, each pod works on several items from the queue and then exits when there are no more items.
+Since the workers themselves detect when the workqueue is empty, and the Job controller does not
+know about the workqueue, it relies on the workers to signal when they are done working.
+The workers signal that the queue is empty by exiting with success.  So, as soon as any worker
+exits with success, the controller knows the work is done, and the Pods will exit soon.
+So, we set the completion count of the Job to 1.  The job controller will wait for the other pods to complete
+too.
+
+
+## Running the Job
+
+So, now run the Job:
+
+```console
+$ kubectl create -f ./job.yaml
+```
+
+Now wait a bit, then check on the job.
+
+```console
+$ ./kubectl describe jobs/job-wq-2 
+Name:		job-wq-2
+Namespace:	default
+Image(s):	gcr.io/causal-jigsaw-637/job-wq-2
+Selector:	app in (job-wq-2)
+Parallelism:	2
+Completions:	1
+Labels:		app=job-wq-2
+Pods Statuses:	0 Running / 1 Succeeded / 0 Failed
+No volumes.
+Events:
+  FirstSeen	LastSeen	Count	From	SubobjectPath	Reason			Message
+  ─────────	────────	─────	────	─────────────	──────			───────
+  1m		1m		1	{job }			SuccessfulCreate	Created pod: job-wq-2-7r7b2
+
+$ kubectl logs pods/job-wq-2-7r7b2
+Worker with sessionID: bbd72d0a-9e5c-4dd6-abf6-416cc267991f
+Inital queue state: empty=False
+Working on banana
+Working on date
+Working on lemon
+```
+
+As you can see, one of our pods worked on several work units.
+
+**TODO:** extend job controller to wait for excess pods to complete, as created via above.
+**TODO(erictune)**: update above example text once issue fixed with Job controller to allow more pods than completions.
+**TODO**: kill a pod or even reboot a node while job is running and have the job still complete.
```

Comment:
If you don't mind I'd rather move those todos as issues, and having this file clean. I'm seeing this as an unnecessary stuff for people going through the examples.